Jenkinsfile

```diff
@@ -275,18 +275,20 @@ def generateStep(version){
   return {
     node('linux && immutable'){
       try {
-        deleteDir()
-        unstash 'source'
         echo "${version}"
-        dir("${BASE_DIR}"){
-          withEnv(["GO_VERSION=${version}"]) {
-            // Another retry in case there are any environmental issues
-            // See https://issuetracker.google.com/issues/146072599 for more context
-            retry(2) {
-              sleep randomNumber(min: 2, max: 5)
+        withEnv(["GO_VERSION=${version}"]) {
+          // Another retry in case there are any environmental issues
+          // See https://issuetracker.google.com/issues/146072599 for more context
+          retry(2) {
+            deleteDir()
+            unstash 'source'
```

Comment:
Retry with deleteDir and unstash to be in a clean context, otherwise:

```
[2020-03-30T08:20:05.898Z] + git clone https://github.com/gin-gonic/gin
[2020-03-30T08:20:05.898Z] fatal: destination path 'gin' already exists and is not an empty directory.
```